soh/src/overlays/actors/ovl_En_Takara_Man/z_en_takara_man.c

```diff
@@ -104,7 +104,14 @@ void func_80B1778C(EnTakaraMan* this, PlayState* play) {
 
         if (!this->unk_21A && this->unk_214) {
             if (Flags_GetSwitch(play, 0x32)) {
-                this->actor.textId = 0x84; //Thanks a lot! (Lost)
+                // text id 0x84 is used in places other than the treasure chest game.
+                // in order to provide a  unique text id that can be replaced for the custom
+                // greg hint, we set it to 0x6E instead
+                if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_GREG_HINT)) {
+                    this->actor.textId = 0x6E;
+                } else {
+                    this->actor.textId = 0x84; //Thanks a lot! (Lost)
+                }
```

Comment:
What does the '0x6E' text ID correspond to? Is it an unused text ID?

Reply:
`0x6E` is the text ID for talking to the treasure shop guy after *winning* the game, it's used on line 100

`0x704C` is the text ID for talking to the treasure shop guy after buying a key but *before* going into the game

Reply:
I think I would've preferred a check for the treasure chest room ID inside OTRGlobals.cpp so all the checks for it would be in one file, but it's not a big deal overall. Thanks for the explanation.